base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -452,19 +456,59 @@ public void dismissKeyboard(@NonNull Integer cancelID) {
      * @param keyboardConfiguration - the custom keyboard configuration to be used when the keyboard is displayed
      */
     public void setKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfiguration) {
```

Comment:
We should call `createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration` whether `keyboardConfiguration` is null or not. Please consider this implemtation for the method 
```
public void setKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfiguration) {
    KeyboardProperties properties = createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration(keyboardConfiguration);
    if (properties == null) {
        this.keyboardConfiguration = defaultKeyboardConfiguration();
    } else {
        this.keyboardConfiguration = properties;
    }
}
```